typescript/src/lib/kurtosis_context/kurtosis_context.ts

```diff
@@ -0,0 +1,184 @@
+                if (error === null) {
+                    if (!response) {
+                        resolve(err(new Error("No error was encountered but the response was still falsy; this should never happen")));
+                    } else {
+                        resolve(ok(response!));
+                    }
+                } else {
+                    resolve(err(error));
+                }
+            })
+        });
+
+        let resultGetEnclaves: Result<GetEnclavesResponse, Error> = await promiseGetEnclaves;
+        if (!resultGetEnclaves.isOk()) {
+            return err(resultGetEnclaves.error)
+        }
+
+        let response: GetEnclavesResponse = resultGetEnclaves.value;
+
+        let enclavesMap: Map<string, EnclaveContext> = this.newEnclaveContextMapFromEnclaveInfoMap(response.getEnclaveInfoMap())
+
+        return ok(enclavesMap)
+    }
+
+    public async destroyEnclave(enclaveId: string): Promise<Result<null, Error>> {
+        const args: DestroyEnclaveArgs = newDestroyEnclaveArgs(enclaveId);
+
+        const promiseDestroyEnclave: Promise<Result<null, Error>> = new Promise((resolve, _unusedReject) => {
+            this.client.destroyEnclave(args, (error: Error | null, _unusedResponse?: google_protobuf_empty_pb.Empty) => {
+                if (error === null) {
+                    resolve(ok(null));
+                } else {
+                    resolve(err(error));
+                }
+            })
+        });
+        const resultDestroyEnclave: Result<null, Error> = await promiseDestroyEnclave;
+        if (!resultDestroyEnclave.isOk()) {
+            return err(resultDestroyEnclave.error);
+        }
+
+        return ok(null);
+    }
+
+    // ====================================================================================================
+    //                                       Private helper functions
+    // ====================================================================================================
+    private newEnclaveContextMapFromEnclaveInfoMap(enclaveInfoMap: jspb.Map<string, EnclaveInfo>): Map<string, EnclaveContext> {
+
+        let enclaveContextMap: Map<string, EnclaveContext> = new Map<string, EnclaveContext>()
+
+        enclaveInfoMap.forEach((value: EnclaveInfo, key: string) => {
+            let enclaveContext: EnclaveContext = this.newEnclaveContextFromEnclaveInfo(value);
+            enclaveContextMap.set(key,enclaveContext);
+        });
+
+        return enclaveContextMap;
+    }
+
+    private newEnclaveContextFromEnclaveInfo(enclaveInfo: EnclaveInfo): EnclaveContext {
+
+        let apiContainerInfo: EnclaveAPIContainerInfo | undefined = enclaveInfo.getApiContainerInfo();
+        let apiContainerContext: APIContainerContext | undefined = undefined;
+
+        let nonExistentApiContainerStatus: number = 0;
+
+        if (enclaveInfo.getApiContainerStatus() != nonExistentApiContainerStatus && apiContainerInfo !== undefined) {
```

Comment:
Same thing from the Go version - no need to do error-checking; we can just do a straight assignment